ProcessLib/HydroMechanics/HydroMechanicsProcess-impl.h

```diff
@@ -0,0 +1,240 @@
+    SecondaryVariableCollection&& secondary_variables,
+    NumLib::NamedFunctionCaller&& named_function_caller)
+    : Process(mesh, std::move(jacobian_assembler), parameters,
+              integration_order, std::move(process_variables),
+              std::move(secondary_variables), std::move(named_function_caller)),
+      _process_data(std::move(process_data))
+{
+}
+
+template <int DisplacementDim>
+bool HydroMechanicsProcess<DisplacementDim>::isLinear() const
+{
+    return false;
+}
+
+template <int DisplacementDim>
+void HydroMechanicsProcess<DisplacementDim>::constructDofTable()
+{
+    // Create single component dof in every of the mesh's nodes.
+    _mesh_subset_all_nodes =
+        std::make_unique<MeshLib::MeshSubset>(_mesh, &_mesh.getNodes());
+    // Create single component dof in the mesh's base nodes.
+    _base_nodes = MeshLib::getBaseNodes(_mesh.getElements());
+    _mesh_subset_base_nodes =
+        std::make_unique<MeshLib::MeshSubset>(_mesh, &_base_nodes);
+
+    // Collect the mesh subsets in a vector.
+
+    // For pressure, which is the first
+    std::vector<MeshLib::MeshSubsets> all_mesh_subsets;
+    all_mesh_subsets.emplace_back(_mesh_subset_base_nodes.get());
+
+    // For displacement.
+    std::generate_n(
+        std::back_inserter(all_mesh_subsets),
+        getProcessVariables()[1].get().getNumberOfComponents(),
+        [&]() { return MeshLib::MeshSubsets{_mesh_subset_all_nodes.get()}; });
+
+    std::vector<unsigned> const vec_n_components{1, DisplacementDim};
+    _local_to_global_index_map =
+        std::make_unique<NumLib::LocalToGlobalIndexMap>(
+            std::move(all_mesh_subsets), vec_n_components,
+            NumLib::ComponentOrder::BY_LOCATION);
+}
+
+template <int DisplacementDim>
+void HydroMechanicsProcess<DisplacementDim>::initializeConcreteProcess(
+    NumLib::LocalToGlobalIndexMap const& dof_table,
+    MeshLib::Mesh const& mesh,
+    unsigned const integration_order)
+{
+    ProcessLib::HydroMechanics::createLocalAssemblers<
+        DisplacementDim, HydroMechanicsLocalAssembler>(
+        mesh.getDimension(), mesh.getElements(), dof_table,
+        // use displacment process variable for shapefunction order
+        getProcessVariables()[1].get().getShapeFunctionOrder(),
+        _local_assemblers, mesh.isAxiallySymmetric(), integration_order,
+        _process_data);
+
+    // TODO move the two data members somewhere else.
+    // for extrapolation of secondary variables
+    std::vector<MeshLib::MeshSubsets> all_mesh_subsets_single_component;
+    all_mesh_subsets_single_component.emplace_back(
+        _mesh_subset_all_nodes.get());
+    _local_to_global_index_map_single_component =
+        std::make_unique<NumLib::LocalToGlobalIndexMap>(
+            std::move(all_mesh_subsets_single_component),
+            // by location order is needed for output
+            NumLib::ComponentOrder::BY_LOCATION);
+
+    Base::_secondary_variables.addSecondaryVariable(
```

Comment:
A comment not on this PR: It could be good that all mechanical related processes share this piece of code.